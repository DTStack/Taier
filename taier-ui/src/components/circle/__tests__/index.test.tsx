import { fireEvent, render } from '@testing-library/react';
import Circle from '..';

describe('Test Circle Component', () => {
    it('Should match snapshots', () => {
        expect(render(<Circle type="running" />).asFragment()).toMatchSnapshot('running');
        expect(render(<Circle type="finished" />).asFragment()).toMatchSnapshot('finished');
        expect(render(<Circle type="stopped" />).asFragment()).toMatchSnapshot('stopped');
        expect(render(<Circle type="frozen" />).asFragment()).toMatchSnapshot('frozen');
        expect(render(<Circle type="fail" />).asFragment()).toMatchSnapshot('fail');
        expect(render(<Circle type="submitting" />).asFragment()).toMatchSnapshot('submitting');
        expect(render(<Circle type="restarting" />).asFragment()).toMatchSnapshot('restarting');
        expect(render(<Circle type="waitSubmit" />).asFragment()).toMatchSnapshot('waitSubmit');
    });

    it('Should call onClick', () => {
        const fn = jest.fn();
        const { getByTestId } = render(<Circle type="running" data-testid="test" onClick={fn} />);
        fireEvent.click(getByTestId('test'));

        expect(fn).toBeCalledTimes(1);
    });
});
