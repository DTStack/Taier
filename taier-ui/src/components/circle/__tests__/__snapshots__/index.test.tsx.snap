// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Circle Component Should match snapshots: fail 1`] = `
<DocumentFragment>
  <div
    class="dtc-circle-default dtc-circle-fail"
  />
</DocumentFragment>
`;

exports[`Test Circle Component Should match snapshots: finished 1`] = `
<DocumentFragment>
  <div
    class="dtc-circle-default dtc-circle-finished"
  />
</DocumentFragment>
`;

exports[`Test Circle Component Should match snapshots: frozen 1`] = `
<DocumentFragment>
  <div
    class="dtc-circle-default dtc-circle-frozen"
  />
</DocumentFragment>
`;

exports[`Test Circle Component Should match snapshots: restarting 1`] = `
<DocumentFragment>
  <div
    class="dtc-circle-default dtc-circle-restarting"
  />
</DocumentFragment>
`;

exports[`Test Circle Component Should match snapshots: running 1`] = `
<DocumentFragment>
  <div
    class="dtc-circle-default dtc-circle-running"
  />
</DocumentFragment>
`;

exports[`Test Circle Component Should match snapshots: stopped 1`] = `
<DocumentFragment>
  <div
    class="dtc-circle-default dtc-circle-stopped"
  />
</DocumentFragment>
`;

exports[`Test Circle Component Should match snapshots: submitting 1`] = `
<DocumentFragment>
  <div
    class="dtc-circle-default dtc-circle-submitting"
  />
</DocumentFragment>
`;

exports[`Test Circle Component Should match snapshots: waitSubmit 1`] = `
<DocumentFragment>
  <div
    class="dtc-circle-default dtc-circle-waitSubmit"
  />
</DocumentFragment>
`;
